client/my-sites/upgrades/domain-management/dns/index.jsx

```diff
@@ -51,9 +67,20 @@ const Dns = React.createClass( {
 						selectedSite={ this.props.selectedSite }
 						selectedDomainName={ this.props.selectedDomainName } />
 
-					<DnsAddNew
-						isSubmittingForm={ this.props.dns.isSubmittingForm }
-						selectedDomainName={ this.props.selectedDomainName } />
+					{ this.state.addNew || this.hasOffice()
+						? <DnsAddNew
+							isSubmittingForm={ this.props.dns.isSubmittingForm }
+							selectedDomainName={ this.props.selectedDomainName } />
+						: <Office365
+							isSubmittingForm={ this.props.dns.isSubmittingForm }
+							selectedDomainName={ this.props.selectedDomainName } />
+					}
+
+					{ ! this.hasOffice()
```

Comment:
Double ternary makes this part hard to understand.

Maybe we can do `{ ! this.hasOffice() && COMPONENT }` instead of `{ ! this.hasOffice() ? COMPONENT : null }`